cli/regex.py

```diff
@@ -1,5 +1,4 @@
 RG_NAME_PATTERN = '^[-\w\._\(\)]{1,90}[^\.]$'
```

Comment:
The use of regex is great!  However you should adopt the commenting style as specified here: 

https://docs.python.org/3/library/re.html#re.VERBOSE

Explaining what each piece of the regex means is super helpful with those not familiar with your pattern.

Reply:
after speaking more with @chrobles, we want to defer this input validation to azure. it's not necessary to do the regex validation checking if azure will already do that for us. we can still put in checks in place to make sure necessary resources are created though